### PR TITLE
Box model

### PR DESCRIPTION
**Pull request by michel2323**
Fix the box model and online checkpointing.

### CODE REVIEW

File: examples/box_model.jl [hunk 1]
```diff
@@ -0,0 +1,195 @@
+#=
```
Comment (sriharikrishna) [inline examples/box_model.jl:1]: I wold remove these commented lines

File: src/Schemes/Online_r2.jl [hunk 2]
```diff
@@ -377,13 +377,16 @@ function checkpoint_struct_while(
             for j= oldcapo:(next_action.iteration-1)
                 if go
                     body(model)
-                    go = condition(model)
+                    onlinesteps=onlinesteps+1
+                else
+                    break
                 end
-                onlinesteps=onlinesteps+1
+                go = condition(model)
```
Comment (sriharikrishna) [inline src/Schemes/Online_r2.jl:384]: The logic here is hard to follow. The assigment`go = condition(model)` can just be placed as the first statement of the loop body and the repeated statement inside the if-true branch can be removed.

**Review by sriharikrishna — COMMENTED** [review]
Some logic may need refining.

**Review by sriharikrishna — COMMENTED** [review]
Some logic needs refining

**Comment by michel2323** [comment]
@sriharikrishna Thank you for the comments!
